docs/Makefile.am

```diff
@@ -25,16 +25,15 @@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 
 if BUILD_DOCS
```

Comment:
Based on @bagder's last point in https://github.com/curl/curl/issues/12921#issuecomment-1937005261, I think this conditional should be dropped. We always want to distribute `mk-ca-bundle.1` and we always want to install and distribute `curl-config.1`.

I'm not sure if we want to add the `all-local` line or not. I mentioned it in the issue, since https://github.com/curl/curl/commit/755b31da10dacca94e0d00029a7306e09e3fe913 said "which also makes it get built by default". If being built upon `make dist` is enough, then it is not needed.

There is already an `EXTRA_DIST` assignment later on in the file, so `mk-ca-bundle.1` should be moved there (or perhaps `CURLPAGES`?), otherwise this one just gets overriden.